cli/src/raffle/trusted_commands/add_raffle.rs

```diff
@@ -0,0 +1,63 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{
+	get_layer_two_nonce,
+	trusted_cli::TrustedCli,
+	trusted_command_utils::{get_identifiers, get_pair_from_str},
+	trusted_operation::perform_trusted_operation,
+	Cli, CliResult, CliResultOk,
+};
+use ita_stf::{Index, RaffleTrustedCall, TrustedCall, WinnerCount};
+use itp_stf_primitives::{
+	traits::TrustedCallSigning,
+	types::{KeyPair, TrustedOperation},
+};
+use itp_types::AccountId;
+use log::*;
+use sp_core::{crypto::Ss58Codec, Pair};
+use std::boxed::Box;
+
+#[derive(Debug, Parser)]
+pub struct AddRaffleCmd {
+	/// Sender's incognito AccountId in ss58check format
+	from: String,
+
+	/// Winner count of the raffle
+	winner_count: WinnerCount,
+}
+
+impl AddRaffleCmd {
+	pub(crate) fn run(&self, cli: &Cli, trusted_args: &TrustedCli) -> CliResult {
```

Comment:
Here we easily see that if we make `CLI` and `TrustedCli` traits, and then put them into core, we can actually move all the raffle commands into `applibs/raffle/cli` without add the commands here, we only need one line of code then to add the commands to the overarching CLI.